core/src/main/java/com/linecorp/armeria/client/retry/RetryRule.java

```diff
@@ -156,26 +170,74 @@ static RetryRule onUnprocessed() {
      * Returns a newly created {@link RetryRuleBuilder}.
      */
     static RetryRuleBuilder builder() {
-        return new RetryRuleBuilder();
+        return builder(HttpMethod.knownMethods());
+    }
+
+    /**
+     * Returns a newly created {@link RetryRuleBuilder} with the specified {@link HttpMethod}s.
+     */
+    static RetryRuleBuilder builder(HttpMethod... methods) {
+        requireNonNull(methods, "methods");
+        return builder(ImmutableSet.copyOf(methods));
+    }
+
+    /**
+     * Returns a newly created {@link RetryRuleBuilder} with the specified {@link HttpMethod}s.
+     */
+    static RetryRuleBuilder builder(Iterable<HttpMethod> methods) {
+        requireNonNull(methods, "methods");
+        final ImmutableSet<HttpMethod> httpMethods = Sets.immutableEnumSet(methods);
+        return builder(headers -> httpMethods.contains(headers.method()));
+    }
+
+    /**
+     * Returns a newly created {@link RetryRuleBuilder} with the specified {@code requestHeadersFilter}.
+     */
+    static RetryRuleBuilder builder(Predicate<? super RequestHeaders> requestHeadersFilter) {
+        return new RetryRuleBuilder(requireNonNull(requestHeadersFilter, "requestHeadersFilter"));
+    }
+
+    /**
+     * Returns a {@link RetryRule} that combines the specified {@code retryRule} and {@code otherRules}.
+     */
+    static RetryRule of(RetryRule retryRule, RetryRule... otherRules) {
+        requireNonNull(retryRule, "retryRule");
```

Comment:
Curious if it's better using a single varargs parameter, because we do so in other places. Could raise an exception on empty rules instead?

Reply:
Yeah, I was considering it too. Let keep consistency.